packages/cross_file/lib/src/types/html.dart

```diff
@@ -64,12 +64,8 @@ class XFile extends XFileBase {
         _lastModified = lastModified ?? DateTime.fromMillisecondsSinceEpoch(0),
         _name = name ?? '',
         super(path) {
-    if (path == null) {
-      _browserBlob = _createBlobFromBytes(bytes, mimeType);
-      _path = URL.createObjectURL(_browserBlob!);
-    } else {
-      _path = path;
-    }
+    _browserBlob = _createBlobFromBytes(bytes, mimeType);
+    _path = path ?? URL.createObjectURL(_browserBlob!);
```

Comment:
This will create a wrong XFile if browserBlob is passed and its path is overridden, down the line, because we've been telling people to use .path on the XFile to access its blob URL; so in this case, if we've created a non empty blob, the path should NOT be used.

I think that in this case, we need to warn users who pass a `path` that it'll be ignored.